src/Message/SNS/Envelope.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Kfz24\QueueBundle\Message\SNS;
+
+use JMS\Serializer\Annotation as Serializer;
+
+/**
+ * Class Envelope
+ * @package Kfz24\QueueBundle\Message\SNS
+ */
+class Envelope
+{
+    /**
+     * @var MessageInterface
+     * @Serializer\MaxDepth(1)
+     */
+    protected $message;
+
+    /**
+     * @var \DateTimeImmutable
+     * @Serializer\Type("DateTimeImmutable")
+     */
+    protected $createdAt;
+
+    /**
+     * AbstractEnvelop constructor.
+     * @param MessageInterface $message
+     */
+    public function __construct(MessageInterface $message)
+    {
+        $this->message = $message;
+    }
+
+    /**
+     * @return int
+     * @Serializer\VirtualProperty("version")
+     */
+    public function getVersion(): int
+    {
+        return $this->message->getVersion();
+    }
```

Comment:
@jacMelloni  Although version is a property of Message, the serialized object is envelope. These getters are proxies that exposes the properties `version`, `type` in the `Envelope` class.

Reply:
Yep I see, it's a very elegant solution and I like it a lot!

But my point was that the `MessageInterface` should also include the `message` field to be exposed

Reply:
If I do that, then the serialized version will be: `[ "version", "type", "message": [ "message", "product_id", "sales_channel_id" ], "created_at" ]`

Reply:
Oh my, that's confusing :smile:

Reply:
Ok you are totally right, I'm gonna go through the code one more time to understand it better!
Thanks :)